gateway-ha/src/main/java/io/trino/gateway/ha/module/HaGatewayProviderModule.java

```diff
@@ -253,8 +271,30 @@ public List<TrinoClusterStatsObserver> getClusterStatsObservers(
     }
 
     @Provides
+    @Singleton
     public MonitorConfiguration getMonitorConfiguration()
     {
         return configuration.getMonitor();
     }
+
+    @Provides
```

Comment:
We don't use `@Provides` for providing simple instances. Please take a look at `BaseApp` class.